passes/sat/sim.cc

```diff
@@ -858,49 +858,53 @@ struct SimInstance
 			Const en = get_state(cell->getPort(ID::EN));
 			Const args = get_state(cell->getPort(ID::ARGS));
 
-			if (!en.as_bool())
-				goto update_print;
-
-			if (trg.size() > 0 && trg_en) {
-				Const trg_pol = cell->getParam(ID::TRG_POLARITY);
-				for (int i = 0; i < trg.size(); i++) {
-					bool pol = trg_pol[i] == State::S1;
-					State curr = trg[i], past = print.past_trg[i];
-					if (pol && curr == State::S1 && past == State::S0)
+			bool sampled = trg_en && trg.size() > 0;
+
+			if (sampled ? print.past_en.as_bool() : en.as_bool()) {
```

Comment:
This got a bit hard to read, I think we should rewrite this (not necessarily in this PR), so that we have:
```
    bool triggered = /* is the cell triggered (not necessarily enabled)? */
    bool enabled = /* EN appropriately sampled */
    if (triggered && enabled)
      /* text rendering */
```